### PR TITLE
fixes #118 NN_TCP_NODELAY doesn't cause TCP_NODELAY to be set

### LINKED ISSUE #118: NN_TCP_NODELAY doesn't cause TCP_NODELAY to be set.

### PR DESCRIPTION
**Pull request by urdh**

### CODE REVIEW

File: src/transports/tcp/atcp.c [hunk 1]
```diff
@@ -192,6 +201,11 @@ static void nn_atcp_handler (struct nn_fsm *self, int src, int type,
                 nn_assert (sz == sizeof (val));
                 nn_usock_setsockopt (&atcp->usock, SOL_SOCKET, SO_RCVBUF,
                     &val, sizeof (val));
+                sz = sizeof (val);
+                nn_ep_getopt (atcp->ep, NN_TCP, NN_TCP_NODELAY, &val, &sz);
+                nn_assert (sz == sizeof (val));
+                nn_usock_setsockopt (&atcp->usock, IPPROTO_TCP, SO_RCVBUF,
```
Comment (gdamore) [inline src/transports/tcp/atcp.c]: This is incorrect.  It is setting the SO_RCVBUF, not the TCP_NODELAY option.   I think you made a copy-pasta error.

**Comment by gdamore** [comment]
Client side code (connect side) looks good.  Server side is using the wrong socket option.  Please fix and retest, then resubmit.  I'd appreciate an understanding of how you validated the test.

**Comment by urdh** [comment]
I wrote and tested (manually observing traffic in Wireshark) the patch for work but reimplemented it at home, that's how the error snuck in there.

It would perhaps be appropriate to have a unit test that inspects the socket options (or something), but I'm not familiar with the nanomsg unit tests, and I don't see any obvious way to do that.

File: src/transports/tcp/atcp.c (hunk 1 above)
Comment (urdh) [inline src/transports/tcp/atcp.c]: Fixed.

**Pull request merged**